test/timing_load_creds.c

```diff
@@ -0,0 +1,196 @@
+{
+    printf("%s %d sec %d microsec\n", what, (int)tp->tv_sec, (int)tp->tv_usec);
+}
+
+static void usage(void)
+{
+    fprintf(stderr, "Usage: %s [flags] pem-file\n", prog);
+    fprintf(stderr, "Flags:\n");
+    fprintf(stderr, "  -c #  Repeat count\n");
+    fprintf(stderr, "  -d    Debugging output (minimal)\n");
+    fprintf(stderr, "  -w<T> What to load T is a single character:\n");
+    fprintf(stderr, "          c for cert (default)\n");
+    fprintf(stderr, "          p for private key\n");
+    exit(1);
+}
+#endif
+
+int main(int ac, char **av)
+{
+#ifndef _WIN32
+    int i, debug = 0, count = 100, what = 'c';
+    struct stat sb;
+    FILE *fp;
+    char *contents;
+    struct rusage start, end, elapsed;
+    struct timeval e_start, e_end, e_elapsed;
+
+    /* Parse JCL. */
+    prog = av[0];
+    while ((i = getopt(ac, av, "c:dw:")) != EOF) {
+        switch (i) {
+        default:
+            usage();
+            break;
+        case 'c':
+            if ((count = atoi(optarg)) < 0)
+                usage();
+            break;
+        case 'd':
+            debug = 1;
+            break;
+        case 'w':
+            if (optarg[1] != '\0')
+                usage();
+            switch (*optarg) {
+            default:
+                usage();
+                break;
+            case 'c':
+            case 'p':
+                what = *optarg;
+                break;
+            }
+            break;
+        }
+    }
+    ac -= optind;
+    av += optind;
+
+    /* Read input file. */
+    if (av[0] == NULL)
+        usage();
+    if (stat(av[0], &sb) < 0) {
+        perror(av[0]);
+        exit(1);
+    }
+    contents = malloc(sb.st_size + 1);
+    if (contents == NULL) {
+        perror("malloc");
+        exit(1);
+    }
+    fp = fopen(av[0], "r");
+    if ((long)fread(contents, 1, sb.st_size, fp) != sb.st_size) {
+        perror("fread");
+        exit(1);
+    }
+    contents[sb.st_size] = '\0';
+    fclose(fp);
+    if (debug)
+        printf(">%s<\n", contents);
+
+    /* Try to prep system cache, etc. */
+    for (i = 10; --i >= 0; ) {
+        switch (what) {
+        case 'c':
+            readx509(contents, (int)sb.st_size);
+            break;
+        case 'p':
+            readpkey(contents, (int)sb.st_size);
+            break;
+        }
+    }
+
+    if (gettimeofday(&e_start, NULL) < 0) {
+        perror("elapsed start");
+        exit(1);
+    }
+    if (getrusage(RUSAGE_SELF, &start) < 0) {
+        perror("start");
+        exit(1);
+    }
+    for (i = count; --i >= 0; ) {
+        switch (what) {
+        case 'c':
+            readx509(contents, (int)sb.st_size);
+            break;
+        case 'p':
+            readpkey(contents, (int)sb.st_size);
+            break;
+        }
+    }
+    if (getrusage(RUSAGE_SELF, &end) < 0) {
+        perror("end");
+        exit(1);
+    }
+    if (gettimeofday(&e_end, NULL) < 0) {
+        perror("end");
```

Comment:
The `perror` before this one also has "end". So, it's hard to differentiate. I find this weird that it's not the actualAPI call that's being printed. But, meh.

Reply:
Changed to refer to the actual calls.